docs/content/en/hugo-pipes/transpilejs.md

```diff
@@ -0,0 +1,69 @@
+---
+title: TranspileJS
+description: Hugo Pipes can process JS files with Babel.
+date: 2019-03-21
+publishdate: 2019-03-21
+lastmod: 2019-03-21
+categories: [asset management]
+keywords: []
+menu:
+  docs:
+    parent: "pipes"
+    weight: 75
+weight: 75
+sections_weight: 75
+draft: false
+---
+
+Any JavaScript resource file can be transpiled to another JavaScript version using `resources.TranspileJS` which takes for argument the resource object and a slice of options listed below. TranspileJS uses the [babel cli](https://babeljs.io/docs/en/babel-cli).
+
+
+{{% note %}}
+Hugo Pipe's TranspileJS requires the `@babel/cli` and `@babel/core` JavaScript packages to be installed in the environment (`npm install -g @babel/cli @babel/core`) along with any Babel plugin(s) or preset(s) used (e.g., `npm install -g @babel/preset-env`).
```

Comment:
I saw the same line in the `postcss` docs, but not sure if this makes sense here (but same applies for postcss). I would never recommend installing this globally and always recommend to install it locally in the project folder (which works btw.). Otherwise if multiple users work on the same repo, they might have different versions. Same if you deploy your website with some service I doubt they all have it installed and if so who knows which version.

But also not sure what the difference is between snap package and whatever this refers to. I use mac and netlify, so do I use snap?

Reply:
I agree as to the recommended best practice for documentation. But there are certain situations where you would want to use the global one (e.g. for my many ad-hoc test sites), but that is, I guess, a limitation of NPM. It puzzles me a little that they have not implemented some kind of global cache (like Go Modules does) to avoid me having to have a million copies of postcss etc. installed.